app/javascript/Components/Result/marks_panel.jsx

```diff
@@ -13,22 +13,45 @@ export class MarksPanel extends React.Component {
     super(props);
     this.state = {
       expanded: new Set(),
+      activeCriterionId: null,
     };
   }
 
+  setActiveCriterion = id => {
+    this.setState({activeCriterionId: id});
+  };
+
+  nextCriterion = () => {
+    const criteria = this.props.marks.filter(
+      c => !(this.props.assigned_criteria !== null && !this.props.assigned_criteria.includes(c.id))
+    ); // skip unassigned
+
+    const idx = criteria.findIndex(c => c.id === this.state.activeCriterionId);
+    const next = idx === -1 || idx === criteria.length - 1 ? criteria[0] : criteria[idx + 1];
```

Comment:
There's another case, when `criteria` is empty (in this case the active criterion should be set to `null`)